### ISSUE TITLE
Add `Vec` support for `Script`s

### ISSUE THREAD

**Issue by digorithm** [FuelLabs/fuels-rs#712]
Arguments passed to a `Script`'s `main` function are being supported once https://github.com/FuelLabs/fuels-rs/pull/620 is merged. 

But, we're leaving `Vec` support for future (hopefully soon) work. This issue is to track this support.

**Comment by iqdecay** [comment]
Closing as duplicate of #655
